operation.go

```diff
@@ -11,6 +12,29 @@ import (
 	"github.com/istreamlabs/huma/schema"
 )
 
+// OperationInfo describes an operation. It contains useful information for
+// logging, metrics, auditing, etc.
+type OperationInfo struct {
+	ID          string
+	URITemplate string
+	Summary     string
+	Tags        []string
+}
+
+// GetOperationInfo returns information about the current Huma operation. This
+// will only be populated *after* routing has been handled, meaning *after*
+// `next.ServeHTTP(w, r)` has been called in your middleware.
+func GetOperationInfo(ctx context.Context) *OperationInfo {
+	if oi := ctx.Value(opIDContextKey); oi != nil {
+		return oi.(*OperationInfo)
+	}
+
+	return &OperationInfo{
+		ID:   "unknown",
+		Tags: []string{},
+	}
```

Comment:
Is there a way to propagate an error somewhere that we can the log/metric on?  It seems to me that no resource ID is an error state and it's something we should probably be tracking, rather than just attaching "unknown" to everything.  In fact we ran into an issue with legacy auditing where we were putting "unknown" in for every request ID for weeks/months and didn't know until we needed that data.

Reply:
I vote we do this a level higher, e.g. at the auditing middleware. It can check for and emit metrics on various states like the operation ID not being present. This low-level functionality just guarantees an object gets returned so you don't have to do nil checks.